core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -56,21 +65,31 @@ public AtomicFileWriter(File f) throws IOException {
     }
 
     /**
-     * @param encoding
-     *      File encoding to write. If null, platform default encoding is chosen.
+     * @param encoding File encoding to write. If null, platform default encoding is chosen.
+     *
+     * @deprecated Use {@link #AtomicFileWriter(File, Charset)}
      */
     public AtomicFileWriter(File f, String encoding) throws IOException {
-        File dir = f.getParentFile();
+        this(f, Charset.forName(encoding));
+    }
+
+        /**
+         * @param charset File charset to write. If null, platform default encoding is chosen.
+         */
+    public AtomicFileWriter(File f, Charset charset) throws IOException {
+        destFile = f.toPath();
+        Path dir = destFile.getParent();
         try {
-            dir.mkdirs();
-            tmpFile = File.createTempFile("atomic",null, dir);
+            if (Files.notExists(dir)) {
+                Files.createDirectories(dir);
```

Comment:
The `if` clause is unnecessary:

> Unlike the `createDirectory` method, an exception is not thrown if the directory could not be created because it already exists.